src/test/java/com/wikia/webdriver/testcases/auth/SignupTests.java

```diff
@@ -0,0 +1,288 @@
+    assertEquals(form.getError(), GENERIC_ERROR_MSG);
+  }
+
+  @Test(groups = DESKTOP)
+  public void userWithSpecialCharactersInUsernameCanSignUpDesktop() {
+    ArticlePageObject article = openArticleOnDesktop();
+    SignUpUser newUser = createNewUserWithSpecialCharacters();
+    signUpOnDesktopAs(newUser);
+    article.verifyUserLoggedIn(newUser.getUsername());
+  }
+
+  @Test(groups = MOBILE)
+  @InBrowser(browser = Browser.CHROME, emulator = Emulator.GOOGLE_NEXUS_5)
+  public void userWithSpecialCharactersInUsernameCanSignUpMobile() {
+    ArticlePage article = openArticleOnMobile();
+    SignUpUser newUser = createNewUserWithSpecialCharacters();
+    signUpOnMobileAs(article, newUser);
+    article.waitForPageReload().verifyUserLoggedIn(newUser.getUsername());
+  }
+
+  @Test(groups = DESKTOP)
+  @Execute(onWikia = MercuryWikis.DISCUSSIONS_2)
+  public void userIsRedirectedToDiscussionPageUponSignUpFromDiscussionPageDesktop() {
+    PostsListPage discussionPage = new PostsListPage().open();
+    signUpOnDesktopFromDiscussionPageAs(createNewUser());
+    assertTrue(discussionPage.waitForPageReload().isStringInURL(PostsListPage.FULL_PATH),
+      "User should be redirected to discussion post list view upon sign up");
+  }
+
+  @Test(groups = MOBILE)
+  @Execute(onWikia = MercuryWikis.DISCUSSIONS_2)
+  @InBrowser(browser = Browser.CHROME, emulator = Emulator.GOOGLE_NEXUS_5)
+  public void userIsRedirectedToDiscussionPageUponSignUpFromDiscussionPageMobile() {
+    PostsListPage discussionPage = new PostsListPage().open();
+    signUpOnDiscussionMobilePageAs(discussionPage, createNewUser());
+    assertTrue(discussionPage.waitForPageReload().isStringInURL(PostsListPage.FULL_PATH),
+      "User should be redirected to discussion post list view upon sign up");
+  }
+
+  @Test(groups = DESKTOP)
+  public void passwordTogglerChangesPasswordVisibilityDesktop() {
+    RegisterPage signUpPage = openSignUpModalOnDesktop();
+    signUpPage.typePassword(existingUser.getPassword());
+
+    assertTrue(signUpPage.isPasswordMasked(), "password should be masked");
+    signUpPage.togglePasswordVisibility();
+    assertFalse(signUpPage.isPasswordMasked(), "password should be readable");
+  }
+
+  @Test(groups = MOBILE)
+  @InBrowser(browser = Browser.CHROME, emulator = Emulator.GOOGLE_NEXUS_5)
+  public void passwordTogglerChangesPasswordVisibilityMobile() {
+    RegisterPage signUpPage = navigateToSignUpOnMobile();
+    signUpPage.typePassword("");
+
+    assertTrue(signUpPage.isPasswordMasked(), "password should be masked");
+    signUpPage.togglePasswordVisibility();
+    assertFalse(signUpPage.isPasswordMasked(), "password should be readable");
+  }
+
+  /**
+   * HELPER METHODS
+   */
+
+  private SignUpUser createNewUser() {
+    return new SignUpUser(
+      String.format("QA%s", Instant.now().getEpochSecond()),
+      getEmailAlias(userWithEmail.getEmail()),
+      String.format("pass_%s", Instant.now().getEpochSecond()),
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Define a constant instead of duplicating this literal "pass_%s" 4 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-s4/coding_rules#rule_key=squid%3AS1192)